test/unit/com/unifina/service/PermissionServiceSpec.groovy

```diff
@@ -66,6 +72,21 @@ class PermissionServiceSpec extends Specification {
 		dashReadPermission = service.grant(anotherUser, dashAllowed, me, Operation.READ)
 		dashAnonymousReadPermission = service.grantAnonymousAccess(anotherUser, dashPublic)
 		service.grant(anotherUser, dashAllowed, anonymousKey)
+
+		streamService = Mock(StreamService)
+		ethereumIntegrationKeyService = Mock(EthereumIntegrationKeyService)
+
+		// Setup application context
+		def applicationContext = Stub(ApplicationContext) {
```

Comment:
This test should extend `BeanMockingSpecification` and use the provided utility methods to register the Mocks as actual beans instead of mocking `ApplicationContext` and `GrailsApplication`